pkg/cloudevents/types/timestamp.go

```diff
@@ -67,17 +60,10 @@ func (t *Timestamp) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
 	if err := d.DecodeElement(&timestamp, &start); err != nil {
 		return err
 	}
-	if pt := ParseTimestamp(timestamp); pt != nil {
-		*t = *pt
-	}
-	return nil
+	var err error
+	*t, err = ParseTimestamp(timestamp)
+	return err
 }
 
 // String outputs the time using layout RFC3339.
-func (t *Timestamp) String() string {
-	if t == nil {
-		return time.Time{}.UTC().Format(time.RFC3339Nano)
-	}
-
-	return t.UTC().Format(time.RFC3339Nano)
-}
+func (t Timestamp) String() string { return t.UTC().Format(time.RFC3339Nano) }
```

Comment:
This should be the canonical form of the time. ToString(t) right?

Reply:
Good point.

Reply:
Fixed, removed all other refs to RFC399 outside value.go, also fixed usage from other packages.